language/go/config.go

```diff
@@ -137,6 +150,13 @@ func newGoConfig() *goConfig {
 		goProtoCompilers: defaultGoProtoCompilers,
 		goGrpcCompilers:  defaultGoGrpcCompilers,
 		goGenerateProto:  true,
+		goEnv: goEnv{
```

Comment:
Would it be better to change the type to a `map[string]string`

```go
goEnv: map[string]string{
  "GOPROXY": "",
  "GONOPROXY": "",
  "GOPRIVATE": "",
  "GOSUMDB": "",
  "GONOSUMDB": "",
}
```

so that you have something you can iterate over when processing? You could go further and make a specific type so that nothing happens on accident.

```go
type goEnvValue string
```